src/com/ms/silverking/cloud/dht/client/impl/SimpleNamespaceCreator.java

```diff
@@ -5,34 +5,34 @@
 
 import com.ms.silverking.cloud.dht.common.Namespace;
 
-public class SimpleNamespaceCreator implements NamespaceCreator {    
-    public SimpleNamespaceCreator() {
-    }
-    
-    @Override
-    public Namespace createNamespace(String namespace) {
-        try {
-            MessageDigest   md;
-            byte[]          bytes;
-            
-            bytes = namespace.getBytes();
-            // FUTURE - think about speeding this up
-            // by using a thread-local digest
-            // like MD5KeyDigest
-            md = MessageDigest.getInstance("MD5");
-            md.update(bytes, 0, bytes.length);
-            return new SimpleNamespace(md.digest());
-        } catch (NoSuchAlgorithmException nsae) {
-            throw new RuntimeException("panic");
-        }
+public class SimpleNamespaceCreator implements NamespaceCreator {
+  public SimpleNamespaceCreator() {
+  }
+
+  @Override
+  public Namespace createNamespace(String namespace) {
+    try {
+      MessageDigest md;
+      byte[] bytes;
+
+      bytes = namespace.getBytes();
+      // FUTURE - think about speeding this up
+      // by using a thread-local digest
+      // like MD5KeyDigest
+      md = MessageDigest.getInstance("MD5");
+      md.update(bytes, 0, bytes.length);
+      return new SimpleNamespace(md.digest());
+    } catch (NoSuchAlgorithmException nsae) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)